pkg/cli/token.go

```diff
@@ -139,30 +146,46 @@ func callOpts(ctx context.Context) ([]grpc.CallOption, error) {
 	return []grpc.CallOption{grpc.PerRPCCredentials(oauth.NewOauthAccess(token))}, nil
 }
 
+// breakglassToken creates a new breakglass token. The token is unsigned.
+//
+// TODO(sethvargo): should this be .NOT_SIGNED or should we sign the token with
+// HMAC using a shared secret exposed via the client.
 func breakglassToken(ctx context.Context, nowUnix int64) (string, error) {
 	now := time.Unix(nowUnix, 0)
-	claims := &jvsapis.JVSClaims{
-		StandardClaims: &jwt.StandardClaims{
-			Audience:  "TODO #22",
-			ExpiresAt: now.Add(ttl).Unix(),
-			Id:        uuid.New().String(),
-			IssuedAt:  now.Unix(),
-			NotBefore: now.Unix(),
-			Issuer:    "jvsctl",
-			Subject:   "jvsctl",
-		},
-		Justifications: []*jvsapis.Justification{{
+	id := uuid.New().String()
+	exp := now.Add(flagTTL)
+	justs := []*jvspb.Justification{
+		{
 			Category: "breakglass",
-			Value:    tokenExplanation,
-		}},
+			Value:    flagTokenExplanation,
+		},
 	}
 
-	// Signing method doesn't really matter because we won't sign
-	// breakglass token.
-	tok, err := jwt.NewWithClaims(jwt.SigningMethodES256, claims).SigningString()
+	token, err := jwt.NewBuilder().
+		Audience([]string{"TODO #22"}).
+		Expiration(exp).
+		IssuedAt(now).
+		Issuer(Issuer).
+		JwtID(id).
+		NotBefore(now).
+		Subject(Subject).
+		Build()
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to build jwt: %w", err)
+	}
+
+	if err := jvspb.SetJustifications(token, justs); err != nil {
+		return "", fmt.Errorf("failed to set justifications on jwt: %w", err)
 	}
 
-	return tok + ".NOT_SIGNED", nil
+	// TODO(sethvargo): the jwt library doesn't actually provide an API for
+	// exposing the raw signing string, so the easiest way to get it is to use the
+	// HMAC signer. We should seriously consider just using an HMAC signer with a
+	// shared secret instead of the hardcoded string.
```

Comment:
We should talk about it in another PR/issue (I don't want to hold this PR up on making that decision, since this change is fully compat with the existing implementation). To support breakglass via HMAC, we'd inspect the JWS envelope and look at the `alg`. If it's `HS256`, we'd parse as breakglass and verify the HMAC. If it's `ES256`, we'd use the JWKs:

```go
// psuedo-code
envelope, err := jws.ParseString(tokenStr, jws.Validate(false))
if err != nil {
  panic(err)
}

var token jwt.Token
switch envelope.Algorithm() {
  case jws.HS256:
    token, err = jwt.ParseString(tokenStr, jws.WithKey(jws.HS256, []byte("jvs")))
  case jws.ES256:
    token, err = jwt.ParseString(tokenStr, jws.WithKeySet(jwksSet, jws.InferAlgorithmFromKey(true))
  default:
    panic("invalid algorithm")
}
```

Those aren't totally the APIs, but I think that gives you the idea.

Reply:
Created: https://github.com/abcxyz/jvs/issues/120. We can discuss it there.